platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ThreadSafeConsensusInfo.java

```diff
@@ -0,0 +1,202 @@
+
+    /**
+     * The minimum judge generation number from the most recent fame-decided round, if there is one.
+     * Else, this is {@link GraphGenerations#FIRST_GENERATION}.
+     *
+     * <p>Updated only on consensus thread, read concurrently from gossip threads.
+     */
+    private volatile long maxRoundGeneration = GraphGenerations.FIRST_GENERATION;
+
+    /**
+     * maximum round number of all events stored in "rounds", or -1 if none. This is the max round
+     * created of all events ever added to the hashgraph.
+     */
+    private volatile long maxRound = ConsensusConstants.ROUND_UNDEFINED;
+    /**
+     * minimum round number of all events stored in "rounds", or -1 if none. This may not be the min
+     * round created of all events ever added to the hashgraph, since some of the older rounds may
+     * have been decided and discarded.
+     */
+    private volatile long minRound = ConsensusConstants.ROUND_UNDEFINED;
+    /** fame has been decided for all rounds less than this, but not for this round. */
+    private volatile long fameDecidedBelow = ConsensusConstants.ROUND_FIRST;
+
+    /**
+     * @param config consensus configuration
+     * @param storage round storage
+     */
+    public ThreadSafeConsensusInfo(
+            @NonNull final ConsensusConfig config, @NonNull final SequentialRingBuffer<MinGenInfo> storage) {
+        this.config = config;
+        this.storage = storage;
+    }
+
+    public @NonNull SequentialRingBuffer<MinGenInfo> getStorage() {
+        return storage;
+    }
+
+    /**
+     * Update all the information stored
+     *
+     * @param fameDecidedBelow the latest round which has not had its fame decided
+     */
+    public void update(final long fameDecidedBelow) {
+        this.fameDecidedBelow = fameDecidedBelow;
+        this.maxRound = storage.maxIndex();
+        this.minRound = storage.minIndex();
+
+        if (fameDecidedBelow == ConsensusConstants.ROUND_FIRST) {
+            // if there are no rounds, set the defaults
+            maxRoundGeneration = GraphGenerations.FIRST_GENERATION;
+            minGenNonAncient = GraphGenerations.FIRST_GENERATION;
+            minRoundGeneration = GraphGenerations.FIRST_GENERATION;
+            return;
+        }
+
+        updateMaxRoundGeneration();
+        updateMinGenNonAncient();
+        updateMinRoundGeneration();
+    }
+
+    /**
+     * Update the max round generation
+     *
+     * <p>Executed only on consensus thread.
+     */
+    private void updateMaxRoundGeneration() {
+        final MinGenInfo info = storage.get(getLastRoundDecided());
+        if (info == null) {
+            // this should never happen
+            LOG.error(
+                    LogMarker.EXCEPTION.getMarker(),
+                    "maxRound({}) is null in updateMaxRoundGeneration()",
+                    getLastRoundDecided());
+            return;
+        }
+        long newMaxRoundGeneration = info.minimumGeneration();
+
+        // 13 May 2021
```

Comment:
Is this dated comment still relevant?

Reply:
updated the comment, still relevant for a short while